File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> n;
        vector<int> res;

        for (int i = 0; i < nums.size(); i++) {
            if (n.contains(target - nums[i])) {
                res.push_back(n[target-nums[i]]);
                res.push_back(i);
                return res;
            }
            n[nums[i]] = i;
        }
        return res;
    }
};

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        h = len(nums) - 1

        while h >= l:
            mid = (h + l) // 2
            if nums[mid] == target:
                return mid
            elif target >= nums[l] and target <= nums[mid]:
                h = mid - 1
            elif target <= nums[h] and target >= nums[mid]:
                l = mid + 1
            elif nums[mid] > nums[h]:
                l = mid + 1
            elif nums[l] > nums[mid]:
                h = mid - 1
            else:
                return -1
            